### PR TITLE
[SPARK-33970][SQL][TEST] Add test default partition in metastoredirectsql 

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This pr add test default partition in metastoredirectsql.

### Why are the changes needed?

Improve test.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by wangyum** [comment]
retest this please.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala [hunk 1]
```diff
@@ -424,6 +428,71 @@ class HivePartitionFilteringSuite(version: String)
...
+      dateStrValue)
+  }
+
+  test("getPartitionsByFilter: d =!= 2019-01-01 || IS NULL") {
+    testMetastorePartitionFiltering(
+      attr("d") =!= Date.valueOf("2019-01-01") || attr("d").isNull,
+      dsValue,
+      hValue,
+      chunkValue,
+      dateValue,
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:472]: @wangyum, why os `2019-01-01` returned here whereas we're using `=!= 2019-01-01`?

Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:472]: Does it return all when null comparison os included?

Comment (wangyum) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:472]: yes.

Comment (wangyum) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:472]: Because it is ||, not &&.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
cc @gatorsmile

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.1 (since this is rather a followup of https://github.com/apache/spark/pull/30534).

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala [hunk 2]
```diff
@@ -39,10 +40,13 @@ class HivePartitionFilteringSuite(version: String)
 
   private val tryDirectSqlKey = HiveConf.ConfVars.METASTORE_TRY_DIRECT_SQL.varname
 
+  // Support default partition in metastoredirectsql since HIVE-11898(Hive 2.0.0).
+  private val defaultPartition = if (version.toDouble >= 2) Some(DEFAULT_PARTITION_NAME) else None
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:44]: If you don't mind, can we use string comparison instead? Something like, `version >= "2.0.0"`?

**Pull request closed without merging**